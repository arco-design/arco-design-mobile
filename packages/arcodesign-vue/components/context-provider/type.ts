export interface ContextInstallOptions {
    /**
     * 组件名称前缀
     * @en Component prefix
     * @default "Arco"
     */
    componentPrefix?: string;
}

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
}
