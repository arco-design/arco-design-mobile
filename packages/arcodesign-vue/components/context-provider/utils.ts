import { InjectionKey, inject } from 'vue';
import { ContextInstallOptions, GlobalContextParams } from './type';

export const defaultInstallOptions: Required<ContextInstallOptions> = {
    componentPrefix: 'Arco',
};

export const defaultContext: Required<GlobalContextParams> = {
    prefixCls: 'arco',
};

export const configProviderInjectionKey: InjectionKey<GlobalContextParams> =
    Symbol('ArcoContextProvider');

export function getComponentPrefix(options?: ContextInstallOptions) {
    return options?.componentPrefix || defaultInstallOptions.componentPrefix;
}

export function getPrefixCls(componentName?: string) {
    const contextProvider = inject(configProviderInjectionKey, defaultContext);
    const prefix = contextProvider.prefixCls;
    if (componentName) {
        return `${prefix}-${componentName}`;
    }
    return prefix;
}
