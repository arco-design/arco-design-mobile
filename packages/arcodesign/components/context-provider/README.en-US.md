### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
