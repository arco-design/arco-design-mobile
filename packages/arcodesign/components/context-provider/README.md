### 其他

# 全局配置 ContextProvider

全局数据控制组件，用于替换全局数据。

======

> 属性/Props

|参数|描述|类型|默认值|
|----------|-------------|------|------|
|prefixCls|组件类名前缀|string|"arco"|
|system|手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|是否使用暗黑模式|boolean|false|
|isDarkMode|是否处于暗黑模式，指定后以指定的值为准|boolean|false|
|theme|主题变量，传入后将在线替换css变量，需设置less变量 @use\-css\-vars: 1|Record\<string, string\>|-|
|locale|国际化语言包配置|ILocale|-|
|useRtl|是否使用Rtl模式|boolean|false|

> ILocale

|参数|描述|类型|
|----------|-------------|------|
|locale|语言类型|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
