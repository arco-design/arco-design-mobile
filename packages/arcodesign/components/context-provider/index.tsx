import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
