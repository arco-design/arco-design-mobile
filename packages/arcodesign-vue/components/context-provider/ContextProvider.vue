<script lang="ts" setup>
import { provide } from 'vue';
import { GlobalContextParams } from './type';
import { configProviderInjectionKey, defaultContext } from './utils';

const props = withDefaults(defineProps<GlobalContextParams>(), defaultContext);
provide(configProviderInjectionKey, props);
</script>
<template>
    <slot />
</template>
