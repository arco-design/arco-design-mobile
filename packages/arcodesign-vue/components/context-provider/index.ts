import ContextProvider from './ContextProvider.vue';
import { installFactory } from '../_helpers/install';

export type * from './type';
export * from './utils';

const install = installFactory(ContextProvider);

export { ContextProvider };

export default {
    install,
};
