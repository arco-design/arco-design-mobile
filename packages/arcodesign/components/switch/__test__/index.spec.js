import React from 'react';
import { mount } from 'enzyme';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import { createMoveTouchEventObject, createStartTouchEventObject } from '../../../tests/helpers/mockEvent';
import { defaultContext } from '../../context-provider';
import Switch from '..';

const prefix = `${defaultContext.prefixCls}-switch`;

demoTest('switch');

mountTest(Switch, 'Switch');

describe('Switch', () => {
    it('should render correctly when set as ios/android', () => {
        const iosComponent = mount(<Switch platform="ios" />);
        expect(iosComponent.find('.type-ios').length).toBe(1);
        const androidComponent = mount(<Switch platform="android" />);
        expect(androidComponent.find('.type-android').length).toBe(1);
    });
    it('should render and callback correctly when did not use checked', () => {
        const component = mount(<Switch />);
        expect(component.find(`.${prefix}`).hasClass('checked')).toBe(false);
        component.simulate('touchStart', createStartTouchEventObject({ x: 100 }));
        component.simulate('touchMove', createMoveTouchEventObject({ x: 100 }));
        component.simulate('touchEnd', createMoveTouchEventObject({ x: 100 }));
        expect(component.find(`.${prefix}`).hasClass('checked')).toBe(true);
    });
    // it('should render and callback correctly when use checked and onChange', () => {
    //     const component = mount(<Switch />);
    //     expect(component.find(`.${prefix}`).hasClass('checked')).toBe(false);
    //     component.simulate('touchStart', createStartTouchEventObject({ x: 100 }));
    //     component.simulate('touchMove', createMoveTouchEventObject({ x: 100 }));
    //     component.simulate('touchEnd', createMoveTouchEventObject({ x: 100 }));
    //     expect(component.find(`.${prefix}`).hasClass('checked')).toBe(true);
    // });
    it('should render and callback correctly when set disabled', () => {
        const mockFn = jest.fn();
        const component = mount(<Switch disabled checked onChange={mockFn} />);
        expect(component.find(`.${prefix}`).hasClass('disabled')).toBe(true);
        expect(component.find(`.${prefix}`).hasClass('checked')).toBe(true);
        component.simulate('touchStart', createStartTouchEventObject({ x: 100 }));
        component.simulate('touchMove', createMoveTouchEventObject({ x: 120 }));
        component.simulate('touchEnd', createMoveTouchEventObject({ x: 130 }));
        expect(component.find(`.${prefix}`).hasClass('checked')).toBe(true);
        expect(mockFn).not.toBeCalled();
    });
    it('should render correctly when use innerArea', () => {

    });
    it('should render correctly when use text', () => {
        const component = mount(<Switch text={{ on: '关', off: '开' }} />);
        expect(component.find(`.${prefix}-text`).text()).toBe('关');
        component.simulate('touchStart', createStartTouchEventObject({ x: 100 }));
        component.simulate('touchMove', createMoveTouchEventObject({ x: 100 }));
        component.simulate('touchEnd', createMoveTouchEventObject({ x: 100 }));
        expect(component.find(`.${prefix}-text`).text()).toBe('开');
    });
    it('should render correctly when use shape', () => {
        const semiComponent = mount(<Switch shape="semi" />);
        expect(semiComponent.find(`.${prefix}`).hasClass('semi')).toBe(true);
        const fullyComponent = mount(<Switch />);
        expect(fullyComponent.find(`.${prefix}`).hasClass('fully')).toBe(true);
    });
    it('should render correctly when use async', () => {

    });
});
