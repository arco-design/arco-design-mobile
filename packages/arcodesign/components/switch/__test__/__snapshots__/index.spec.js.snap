// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch demo test switch demo: index.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body"
    >
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              IOS ON/OFF
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              aria-checked="true"
              aria-disabled="false"
              class="arco-switch all-border-box type-ios system-pc fully checked"
              role="switch"
              tabindex="0"
            >
              <div
                class="arco-switch-inner"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              IOS ON/Disabled
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              aria-checked="true"
              aria-disabled="true"
              class="arco-switch all-border-box type-ios system-pc fully checked disabled"
              role="switch"
              tabindex="0"
            >
              <div
                class="arco-switch-inner"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              IOS OFF/Disabled
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              aria-checked="false"
              aria-disabled="true"
              class="arco-switch all-border-box type-ios system-pc fully disabled"
              role="switch"
              tabindex="0"
            >
              <div
                class="arco-switch-inner"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`switch demo test switch demo: index2.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body"
    >
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              Android ON/OFF
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              aria-checked="true"
              aria-disabled="false"
              class="arco-switch all-border-box type-android system-pc fully checked"
              role="switch"
              tabindex="0"
            >
              <div
                class="arco-switch-inner"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              Android ON/Disabled
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              aria-checked="true"
              aria-disabled="true"
              class="arco-switch all-border-box type-android system-pc fully checked disabled"
              role="switch"
              tabindex="0"
            >
              <div
                class="arco-switch-inner"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              Android OFF/Disabled
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              aria-checked="false"
              aria-disabled="true"
              class="arco-switch all-border-box type-android system-pc fully disabled"
              role="switch"
              tabindex="0"
            >
              <div
                class="arco-switch-inner"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`switch demo test switch demo: index3.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell all-border-box without-group"
  >
    <div
      class="arco-cell-inner"
    >
      <div
        class="cell-label"
      >
        <div
          class="cell-title"
        >
          Loading Status
        </div>
      </div>
      <div
        class="cell-content has-label"
      >
        <div
          aria-checked="true"
          aria-disabled="false"
          class="arco-switch all-border-box demo-3 type-ios system-pc fully checked"
          role="switch"
          tabindex="0"
        >
          <div
            class="arco-switch-inner"
          >
            <div
              class="arco-loading all-border-box circle"
              style="animation-duration: 1000ms; width: 14px; height: 14px;"
            >
              <svg
                viewBox="0 0 14 14"
              >
                <defs>
                  <lineargradient
                    id="grad1-inner-0"
                    x1="0%"
                    x2="100%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      class="loading-circle-middle stop-color-with-config"
                      offset="0%"
                    />
                    <stop
                      class="loading-circle-start stop-color-with-config"
                      offset="100%"
                    />
                  </lineargradient>
                  <lineargradient
                    id="grad2-inner-0"
                    x1="0%"
                    x2="100%"
                    y1="0%"
                    y2="0%"
                  >
                    <stop
                      class="loading-circle-middle stop-color-with-config"
                      offset="0%"
                    />
                    <stop
                      class="loading-circle-end stop-color-with-config"
                      offset="100%"
                    />
                  </lineargradient>
                </defs>
                <circle
                  cx="7"
                  cy="7"
                  fill="none"
                  r="6"
                  stroke="url(#grad1-inner-0)"
                  stroke-dasharray="18.84955592153876"
                  stroke-dashoffset="18.84955592153876"
                  stroke-width="2"
                />
                <circle
                  cx="7"
                  cy="7"
                  fill="none"
                  r="6"
                  stroke="url(#grad2-inner-0)"
                  stroke-dasharray="18.84955592153876"
                  stroke-width="2"
                />
                <circle
                  class="loading-circle-filleted fill-color-with-config"
                  cx="13"
                  cy="7"
                  r="1"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`switch demo test switch demo: index4.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell all-border-box without-group"
  >
    <div
      class="arco-cell-inner"
    >
      <div
        class="cell-label"
      >
        <div
          class="cell-title"
        >
          Open/Close
        </div>
      </div>
      <div
        class="cell-content has-label"
      >
        <div
          aria-checked="false"
          aria-disabled="false"
          class="arco-switch all-border-box demo-4 type-ios system-pc fully"
          role="switch"
          tabindex="0"
        >
          <span
            class="arco-switch-text"
          >
            ON
          </span>
          <div
            class="arco-switch-inner"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`switch demo test switch demo: index5.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell all-border-box without-group"
  >
    <div
      class="arco-cell-inner"
    >
      <div
        class="cell-label"
      >
        <div
          class="cell-title"
        >
          Android Right Angle
        </div>
      </div>
      <div
        class="cell-content has-label"
      >
        <div
          aria-checked="false"
          aria-disabled="false"
          class="arco-switch all-border-box type-android system-pc semi"
          role="switch"
          tabindex="0"
        >
          <div
            class="arco-switch-inner"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`switch demo test switch demo: index6.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell all-border-box without-group"
  >
    <div
      class="arco-cell-inner"
    >
      <div
        class="cell-label"
      >
        <div
          class="cell-title"
        >
          Asynchronous control, secondary confirmation
        </div>
      </div>
      <div
        class="cell-content has-label"
      >
        <div
          aria-checked="false"
          aria-disabled="false"
          class="arco-switch all-border-box type-ios system-pc fully"
          role="switch"
          tabindex="0"
        >
          <div
            class="arco-switch-inner"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
